lib/bitcoin/SpendingMonitor.ts

```diff
@@ -0,0 +1,113 @@
+import ITransactionStore from '../core/interfaces/ITransactionStore';
+import TransactionModel from '../common/models/TransactionModel';
+import TransactionNumber from './TransactionNumber';
+
+/**
+ * Encapsulates the functionality to calculate the amount of spending that the
+ * service is doing.
+ */
+export default class SpendingMonitor {
+
+  // This is the max number of results that we want from the transaction store
+  private static readonly maxResultsFromTransactionStore = 10000;
+
+  private static readonly estimatedBlockWritesInOneMonth = 6 * 24 * 30;
+
+  private anchorStringsWritten: Set<string>;
+
+  public constructor (
+    private bitcoinFeeSpendingCutoffPeriodInBlocks: number,
+    private bitcoinFeeSpendingCutoffInSatoshis: number,
+    private transactionStore: ITransactionStore) {
+
+    if (bitcoinFeeSpendingCutoffPeriodInBlocks < 1 || bitcoinFeeSpendingCutoffPeriodInBlocks > SpendingMonitor.estimatedBlockWritesInOneMonth) {
+      // tslint:disable-next-line: max-line-length
+      throw new Error(`Bitcoin spending cutoff period: ${bitcoinFeeSpendingCutoffPeriodInBlocks} must be between 1-${SpendingMonitor.estimatedBlockWritesInOneMonth}`);
+    }
+
+    if (bitcoinFeeSpendingCutoffInSatoshis <= 0) {
+      throw new Error('Bitcoin spending cutoff amount must be > 0');
+    }
+
+    this.anchorStringsWritten = new Set<string>();
+  }
+
+  /**
+   * Add the transaction data to track the spending.
+   * @param anchorString The string to be written.
+   */
+  public addTransactionDataBeingWritten (anchorString: string): void {
+    this.anchorStringsWritten.add(anchorString);
+  }
+
+  /**
+   * Calculates whether the specified fee will result in this node going over the spending limit.
+   * @param currentFeeInSatoshis The fee to be added for the next transaction.
+   * @param startingBlockHeight The block height to start the check for the cutoff period.
+   */
+  public async isCurrentFeeOverSpendingLimit (currentFeeInSatoshis: number, lastProcessedBlockHeight: number): Promise<boolean> {

```

Comment:
Style (minor thus optional), consider having returned `true` as the "happy" state as opposed to the "failure" state, so consider `isCurrentFeeWithinSpendingLimit`.